include/unifex/scope_guard.hpp

```diff
@@ -16,27 +16,41 @@
 #pragma once
 
 #include <type_traits>
+#include <utility>
 
 #include <unifex/detail/prologue.hpp>
 
 namespace unifex {
 
 template <typename Func>
 struct scope_guard {
+private:
+  static_assert(std::is_nothrow_move_constructible_v<Func>);
   UNIFEX_NO_UNIQUE_ADDRESS Func func_;
+  bool dismissed_ = false;
 
-  scope_guard(Func&& func) noexcept(
-      std::is_nothrow_move_constructible_v<Func>)
+public:
+  scope_guard(Func&& func) noexcept
       : func_((Func &&) func) {}
 
   ~scope_guard() {
+    reset();
+  }
+
+  void dismiss() {
+    dismissed_ = true;
+  }
+
+  void reset() {
```

Comment:
`noexcept`?

Reply:
The name `reset()` doesn't really scream "run this now" to me when I read it at the call-site. I suspect there would be occasions where I would mistake `.reset()` for doing the same thing as `.dismiss()`.

Reply:
Maybe `dismiss` should be `release` to more strongly evoke the `reset`/`release` of standard smart pointers.